lib/locations_ng/city.rb

```diff
@@ -1,31 +1,32 @@
 module LocationsNg
   class City
+    @@all_cities
+
     class << self
       def all
         load_cities
       end
 
-      def cities(state)
-        state_query = state.downcase.gsub(' ', '_')
-        all_cities = load_cities
+      def state_query(state)
+        state == 'federal_capital_territory' ? 'fct' : state
+      end
 
-        if state_query == 'federal_capital_territory'
-          state_query = 'fct'
-        end
+      def cities(state)
+        load_cities
+        state_query = self.state_query(state.downcase.gsub(' ', '_'))
 
-        city_index = all_cities.index{|c| c['alias'] == state_query}
+        city_index = @@all_cities.index{ |c| c['alias'] == state_query }
 
-        if city_index.nil?
-          {message: "No cities found for '#{state}'", status: 404}
-        else
-          all_cities[city_index]['cities']
+        unless city_index.nil?
+          return @@all_cities[city_index]['cities']
         end
+
+        {message: "No cities found for '#{state}'", status: 404}
       end
 
       private
-
       def load_cities
-        YAML.load(File.read(files_location 'cities'))
+        @@all_cities ||= YAML.load(File.read(files_location 'cities'))
```

Comment:
Good idea. I will move this to a separate class in the weekend and see if there are any performance gains. I will let you know how that goes.